pennylane/labs/dla/variational_kak.py

```diff
@@ -0,0 +1,397 @@
+    and a Hermitian operator :math:`H \in \tilde{\mathfrak{m}} \oplus \mathfrak{a}`, this function computes
+    :math:`a \in \mathfrak{a}` and :math:`K_c \in e^{i\mathfrak{k}}` such that
+
+    .. math:: H = K_c a K_c^\dagger
+
+    The result is provided in terms of the adjoint vector representation of :math:`a \in \mathfrak{a}`
+    (see :func:`adjvec_to_op`), i.e. the ordered coefficients :math:`c_j` in :math:`a = \sum_j c_j m_j` with the basis elements :math:`m_j \in (\tilde{\mathfrak{m}} \oplus \mathfrak{a})` and
+    the optimal parameters :math:`\theta` such that
+
+    .. math:: K_c = \prod_{j=|\mathfrak{k}|}^{1} e^{-i \theta_j k_j}
+
+    for the ordered basis of :math:`\mathfrak{k}` given by the first :math:`|\mathfrak{k}|` elements of ``g``.
+    Note that we define :math:`K_c` mathematically with the descending order of basis elements :math:`k_j in \mathfrak{k}` such that
+    the resulting circuit has the canonical ascending order. In particular a PennyLane quantum function that describes the circuit given
+    the optimal parameters ``theta_opt`` and the basis ``k`` containing the operators, is given by the following.
+
+    .. code-block::
+        def Kc(theta_opt: Iterable[float], k: Iterable[Operator]):
+            assert len(theta_opt) == len(k)
+            for theta_j, k_j in zip(theta_opt, k):
+                qml.exp(-1j * theta_j * k_j)
+
+    Internally, this function performs a modified version of `2104.00728 <https://arxiv.org/abs/2104.00728>`__,
+    in particular minimizing the cost function
+
+    .. math:: f(\theta) = \langle H, K(\theta) e^{-i \sum_{j=1}^{|\mathfrak{a}|} \pi^j a_j} K(\theta)^\dagger \rangle,
+
+    see eq. (6) therein and our `demo <https://pennylane.ai/qml/demos/tutorial_fdhs>`__ for more details.
+    Instead of relying on having Pauli words, we use the adjoint representation
+    for a more general evaluation of the cost function. The rest is the same.
+
+    .. seealso:: `Theory demo on KAK theorem <https://pennylane.ai/qml/demos/tutorial_kak_theorem>`__, `demo on KAK decomposition in practise <https://pennylane.ai/qml/demos/tutorial_fdhs>`__,
+
+    Args:
+        H (Union[Operator, PauliSentence, np.ndarray]): Hamiltonian to decompose
+        g (List[Union[Operator, PauliSentence, np.ndarray]]): DLA of the Hamiltonian
+        dims (Tuple[int]): Tuple of dimensions ``(dim_k, dim_mtilde, dim_a)`` of
+            Cartan decomposition :math:`\mathfrak{g} = \mathfrak{k} \oplus (\tilde{\mathfrak{m}} \oplus \mathfrak{a})`
+        adj (np.ndarray): Adjoint representation of dimension ``(dim_g, dim_g, dim_g)``,
+            with the implicit ordering ``(k, mtilde, a)``.
+        verbose (bool): Plot the optimization
+        opt_kwargs (dict): Keyword arguments for the optimization like initial starting values for :math:`\theta` of dimension ``(dim_k,)``.
+            Also includes ``n_epochs``, ``lr``, ``b1``, ``b2``, ``verbose``, ``interrupt_tol``, see :func:`~run_opt`
+
```

Comment:
```suggestion
        pick_min (bool): Whether to pick the parameter set with lowest cost function value during the optimization
            as optimal parameters. Otherwise picks the last parameter set.
```